src/test/java/lotto/domain/LottoCustomerTest.java

```diff
@@ -0,0 +1,95 @@
+package lotto.domain;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+
+class LottoCustomerTest {
+
+    private static LottoTicket lottoTicket1;
+    private static LottoTicket lottoTicket2;
+    private static LottoCustomer customer;
+
+    @BeforeEach
+    void setup() {
+        lottoTicket1 = new LottoTicket(Money.wons(100), LottoNumbers.of(1, 2, 3, 4, 5, 6));
+        lottoTicket2 = new LottoTicket(Money.wons(200), LottoNumbers.of(45, 44, 43, 42, 41, 40));
+        customer = new LottoCustomer(Money.wons(1_000));
+    }
+
+    @DisplayName("가진 금액이, 로또 구매 가격보다 적다면, 로또를 구입 할 수 없다")
+    @ParameterizedTest
+    @ValueSource(ints = {0, 50, 99})
+    void purchase_nothing(final int amount) {
+        final Money initialAmount = Money.wons(amount);
+        final LottoCustomer customer = new LottoCustomer(initialAmount);
+
+        // Validate PreCondition
+        assertThat(customer.canPurchase(lottoTicket1)).isFalse();
+
+        assertThat(customer.purchase(lottoTicket1)).isEqualTo(Money.ZERO);
+
+        // Validate PostCondition
+        assertThat(customer.getPurchasedCount()).isZero();
+        assertThat(customer.getPurchasedLottoTickets()).isEmpty();
+    }
+
+    @DisplayName("가진 금액이, 로또를 딱 한 번 구매 가능한 금액이라면, 로또를 한 번 구입할 수 있다")
+    @ParameterizedTest
+    @ValueSource(ints = {100, 101, 150, 199})
+    void purchase_onlyOne(final int amount) {
+        final Money initialAmount = Money.wons(amount);
+        final LottoCustomer customer = new LottoCustomer(initialAmount);
+
+        // Validate PreCondition
+        assertThat(customer.canPurchase(lottoTicket1)).isTrue();
+        assertThat(customer.getPurchasedCount()).isEqualTo(0);
+
+        assertThat(customer.purchase(lottoTicket1)).isEqualTo(lottoTicket1.getFee());
+
+        // Validate PostCondition
+        assertThat(customer.canPurchase(lottoTicket1)).isFalse();
+        assertThat(customer.getPurchasedCount()).isEqualTo(1);
+        assertThat(customer.getPurchasedLottoTickets()).containsExactly(lottoTicket1);
+    }
+
+    @DisplayName("가진 금액으로, 구입 가능한 만큼, 로또를 구입 할 수 있다")
+    @ParameterizedTest
+    @ValueSource(ints = {1, 5, 10, 100})
+    void purchase_multiple(final int purchaseCount) {
+        final Money initialAmount = lottoTicket1.getFee().multiply(Money.wons(purchaseCount));
+        final LottoCustomer customer = new LottoCustomer(initialAmount);
+
+        while (customer.canPurchase(lottoTicket1)) {
+            customer.purchase(lottoTicket1);
+        }
```

Comment:
테스트 코드안에 프로덕션 코드가 있는것 같네요 :thinking: 

프로덕션 코드가 테스트 코드내에 있을경우, 프로덕션 코드가 변경되었을때 테스트 코드또한 같이 관리해야하는 이슈가 발생해서 개인적으로 지양하는 편입니다.

그렇다면, `가진 금액만큼 로또를 구입한다.` 이러한 메소드가 있고 그러한 메서드를 테스트 하는게 좋을것 같다는 생각입니다 :pray: 

```suggestion
        final Money initialAmount = lottoTicket1.getFee().multiply(Money.wons(purchaseCount));
        final LottoCustomer customer = new LottoCustomer(initialAmount);

        customer.buyAllLotto();
```

예시를 들면, 저는 이렇게 진행했을것 같아요.

1. 프로덕션 코드를 구현하기전에 테스트 코드를 구현한다.
```
void test() {
 int a = 1;
 int b = 2; 
 
 int result = a + b;

 검증(result, 3);
}
```
2. 테스트 코드를 프로덕션 코드로 옮긴다.
```
void test() {
  int a = 1;
  int b = 2; 
  Calculate calculate = new Calculate();

  int result = calculate.sum(a,b);

  검증(result, 3);
}
```

---

> 유닛 테스트(unit test)는 컴퓨터 프로그래밍에서 소스 코드의 특정 모듈이 의도된 대로 정확히 작동하는지 검증하는 절차다. 즉, 모든 함수와 메소드에 대한 테스트 케이스(Test case)를 작성하는 절차를 말한다.
[위키백과](https://ko.wikipedia.org/wiki/%EC%9C%A0%EB%8B%9B_%ED%85%8C%EC%8A%A4%ED%8A%B8)

Reply:
피드백 감사합니다 :)
궁금한 점 질문드릴게요!

> 프로덕션 코드가 테스트 코드내에 있을경우, 프로덕션 코드가 변경되었을때 테스트 코드또한 같이 관리해야하는 이슈가 발생해서 개인적으로 지양하는 편입니다.

-> 테스트 코드란,소스 코드의 특정 모듈인 프로덕션 코드또한 의도대로 동작하는지를 검증하기 위해 존재한다고 개인적으로 생각합니다만, 이 부분에 대해서 어떻게 생각하시나요?
(따라서 프로덕션 코드가 변경되었을때는 당연히 테스트 코드 또한 변경되야 된다고 생각합니다)